framework/src/wifi_manager/wifi_manager.c

```diff
@@ -722,41 +790,56 @@ wifi_manager_result_e _handler_on_connecting_state(_wifimgr_msg_s *msg)
 wifi_manager_result_e _handler_on_connected_state(_wifimgr_msg_s *msg)
 {
 	WM_LOG_HANDLER_START;
-	wifi_manager_result_e res;
 	if (msg->event == EVT_DISCONNECT) {
-		// call disconnect_ap
-		res = _wifimgr_disconnect_ap();
-		if (res != WIFI_MANAGER_SUCCESS) {
-			// this should succeed
-		}
+		WIFIMGR_CHECK_RESULT(_wifimgr_disconnect_ap(), "critical error", WIFI_MANAGER_FAIL);
 		WIFIMGR_SET_STATE(WIFIMGR_STA_DISCONNECTING);
 	} else if (msg->event == EVT_STA_DISCONNECTED) {
-		// ap is disconnected from remote
+		/* ap is disconnected from remote */
 		_handle_user_cb(CB_STA_DISCONNECTED, NULL);
-		WIFIMGR_SET_STATE(WIFIMGR_STA_DISCONNECTED);
+		if (g_manager_info.conn_config.type == WIFI_RECONN_NONE) {
+			WIFIMGR_SET_STATE(WIFIMGR_STA_DISCONNECTED);
+		} else {
+			_wifimgr_conn_info_msg_s *rmsg = (_wifimgr_conn_info_msg_s *)malloc(sizeof(_wifimgr_conn_info_msg_s));
+			if (!rmsg) {				// critical error
+				ndbg("allocate memory for reconn msg fail\n");
+				return WIFI_MANAGER_FAIL;
+			}
+			rmsg->config = (wifi_manager_ap_config_s *)malloc(sizeof(wifi_manager_ap_config_s));
+			if (!rmsg->config) {				// critical error
+				free(rmsg);
+				ndbg("allocate memory for reconnmsg apconfig is fail\n");
+				return WIFI_MANAGER_FAIL;
+			}
+
+			rmsg->conn_config = (wifi_manager_connect_config_s *)malloc(sizeof(wifi_manager_connect_config_s));
+			if (!rmsg->conn_config) {				// critical error
+				free(rmsg->config);
+				free(rmsg);
+				ndbg("allocate memory for reonnmsg conn_config is fail\n");
+				return WIFI_MANAGER_FAIL;
+			}
+			WIFIMGR_COPY_AP_INFO(*(rmsg->config), g_manager_info.connected_ap);
```

Comment:
why didn't use a pointer instead of memory copy even though g_manager_info is protected?

Reply:
it's not a function but macro. so it doesn't care